Remove page refresh after move

jonmast opened the pull request:
This pushes json down to the client which is used to update the board
client-side. Still need to add support for status messages as well as
special moves (castling, en passant, pawn promotion, etc.)

jonmast committed:
Remove page refresh after move

This pushes json down to the client which is used to update the board
client-side. Still need to add support for status messages as well as
special moves (castling, en passant, pawn promotion, etc.)

radiofreemattd commented on app/assets/javascripts/games.js.coffee:
```diff
@@ -1,12 +1,16 @@
+      if piece.css('top') == '0px' && piece.css('left') == '0px'
+        animateMove(piece, data.destination)
+      else
+        movePieceElement(piece, data.destination)
```
This looks good...help me understand this `if` block better?

jonmast replied:
When the user drags a piece it updates the top and left offsets to move the piece. If they are both zero we can assume the piece has not been dragged and therefore we want to animate it into position. Essentially the second case is for the user that made the move and the other case is for everyone else. I'm definitely open to suggestions/patches to make this code clearer.   My JS tends to spaghettify rather quickly.

jonmast committed:
Update status messages in realtime

Add an additional pusher event which pushes down a status message. Also
add z-index to pieces when dragging so they stay on top. Fixes #96.
